test/box-luatest/rv_txs_conflict_on_dml_test.lua

```diff
@@ -0,0 +1,45 @@
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+
+local pg = t.group(nil, t.helpers.matrix({engine = {'memtx', 'vinyl'},
+                                          op = {'replace', 'insert', 'delete',
+                                                'upsert'}}))
```

Comment:
1. No need to add two test files for this - you can define more than one group in the same file.
2. Please add a brief comment explaining what each test case expects.
3. Please add a test case ensuring that no error is thrown if the conflicted transaction is read only.
4. Please don't use `eval`. Use `exec` for better code readability. You can pass arguments to `exec` like this:
   ```Lua
   g.server:exec(function(arg1, arg2) end, {arg1, arg2})
   ```
5. It isn't necessary to restart the server after each test case. All you need to do is drop the space and reopen the streams. For a test case to leave no traces you could add `before_each` and `after_each` callbacks that would open and close/rollback the streams.
6. Since the test is multi-engine, it should probably be located in `test/engine-luatest`.

Reply:
1. As I split the change log into two, I decided to not merge these test and rename to indicate the issues they test (#7238 and #7239). #7240 is basically a declaration of the breaking change needed to fix the former tickets — that's why there is no separate test for it.
2. Added brief comments for each test — how about checking for this in checkpatch?:wink:
3. Added this test case.
4. Thanks for pointing this out! I didn't even know one could pass arguments to `eval` closures — clearly, this is not even mentioned in **luatest**. I got rid of `eval` everywhere except for this spot:
https://github.com/CuriousGeorgiy/tarantool/blob/64b18c1d66313871dec35facca575a3517383556/test/engine-luatest/gh_7238_memtx_fix_conflicted_txs_ro_operations_test.lua#L49-L52

For some reason `pairs` is not supported in `net.box`, so I cannot test read-only methods in a generic way.
5. Also thank you for pointing this out to me! I blindly copy-pasted this from replication tests that use parametrization groups and was not mindful enough to question why it is done this way there. Fixed: moved this logic to `before_all` and `after_all` and added necessary clean up to `before_each` and `after_each`. AFAIC, streams are automatically closed when their corresponding **Lua** objects get garbage collected, aren't they?
6. Fixed: move these tests to `test/engine-luatest`.